From 41c86f2d2d51f4481ab13bcebf5f453b3dd722ee Mon Sep 17 00:00:00 2001
From: Pekka Paalanen <pekka.paalanen@collabora.com>
Date: Thu, 4 Apr 2019 17:29:27 +0300
Subject: [PATCH 16/45] libweston: move gl-renderer into a subdir

GL-renderer is expected to grow more files, both by addition and by splitting.
Moving them into a new subdirectory helps people to understand which files are
relevant.

Signed-off-by: Pekka Paalanen <pekka.paalanen@collabora.com>
---
 libweston/compositor-drm.c                |  2 +-
 libweston/compositor-wayland.c            |  2 +-
 libweston/compositor-x11.c                |  2 +-
 libweston/meson.build                     | 38 ++---------------------
 libweston/{ => renderer-gl}/gl-renderer.c |  0
 libweston/{ => renderer-gl}/gl-renderer.h |  0
 libweston/renderer-gl/meson.build         | 37 ++++++++++++++++++++++
 7 files changed, 42 insertions(+), 39 deletions(-)
 rename libweston/{ => renderer-gl}/gl-renderer.c (100%)
 rename libweston/{ => renderer-gl}/gl-renderer.h (100%)
 create mode 100644 libweston/renderer-gl/meson.build

diff --git a/libweston/compositor-drm.c b/libweston/compositor-drm.c
index 7fba893f..136acb74 100644
--- a/libweston/compositor-drm.c
+++ b/libweston/compositor-drm.c
@@ -55,7 +55,7 @@
 #include "weston-debug.h"
 #include "shared/helpers.h"
 #include "shared/timespec-util.h"
-#include "gl-renderer.h"
+#include "renderer-gl/gl-renderer.h"
 #include "weston-egl-ext.h"
 #include "pixman-renderer.h"
 #include "pixel-formats.h"
diff --git a/libweston/compositor-wayland.c b/libweston/compositor-wayland.c
index 7e2fee17..337ea105 100644
--- a/libweston/compositor-wayland.c
+++ b/libweston/compositor-wayland.c
@@ -46,7 +46,7 @@
 
 #include <libweston/libweston.h>
 #include <libweston/backend-wayland.h>
-#include "gl-renderer.h"
+#include "renderer-gl/gl-renderer.h"
 #include "weston-egl-ext.h"
 #include "pixman-renderer.h"
 #include "shared/helpers.h"
diff --git a/libweston/compositor-x11.c b/libweston/compositor-x11.c
index 3fdeb114..854d053f 100644
--- a/libweston/compositor-x11.c
+++ b/libweston/compositor-x11.c
@@ -56,7 +56,7 @@
 #include "shared/image-loader.h"
 #include "shared/timespec-util.h"
 #include "shared/file-util.h"
-#include "gl-renderer.h"
+#include "renderer-gl/gl-renderer.h"
 #include "weston-egl-ext.h"
 #include "pixman-renderer.h"
 #include "presentation-time-server-protocol.h"
diff --git a/libweston/meson.build b/libweston/meson.build
index c93307dc..b04b1968 100644
--- a/libweston/meson.build
+++ b/libweston/meson.build
@@ -426,42 +426,6 @@ dep_vertex_clipping = declare_dependency(
 	include_directories: include_directories('.')
 )
 
-if get_option('renderer-gl')
-	config_h.set('ENABLE_EGL', '1')
-
-	srcs_renderer_gl = [
-		'gl-renderer.c',
-		linux_dmabuf_unstable_v1_protocol_c,
-		linux_dmabuf_unstable_v1_server_protocol_h,
-	]
-
-	deps_renderer_gl = [
-		dep_pixman,
-		dep_libweston,
-		dep_libdrm_headers,
-		dep_vertex_clipping
-	]
-
-	foreach name : [ 'egl', 'glesv2' ]
-		d = dependency(name, required: false)
-		if not d.found()
-			error('gl-renderer requires @0@ which was not found. Or, you can use \'-Drenderer-gl=false\'.'.format(name))
-		endif
-		deps_renderer_gl += d
-	endforeach
-
-	plugin_gl = shared_library(
-		'gl-renderer',
-		srcs_renderer_gl,
-		include_directories: include_directories('..', '../shared'),
-		dependencies: deps_renderer_gl,
-		name_prefix: '',
-		install: true,
-		install_dir: dir_module_libweston
-	)
-	env_modmap += 'gl-renderer.so=@0@;'.format(plugin_gl.full_path())
-endif
-
 if get_option('weston-launch')
 	dep_pam = cc.find_library('pam')
 
@@ -479,3 +443,5 @@ if get_option('weston-launch')
 
 	meson.add_install_script('echo', 'REMINDER: You are installing weston-launch, please make it setuid-root.')
 endif
+
+subdir('renderer-gl')
diff --git a/libweston/gl-renderer.c b/libweston/renderer-gl/gl-renderer.c
similarity index 100%
rename from libweston/gl-renderer.c
rename to libweston/renderer-gl/gl-renderer.c
diff --git a/libweston/gl-renderer.h b/libweston/renderer-gl/gl-renderer.h
similarity index 100%
rename from libweston/gl-renderer.h
rename to libweston/renderer-gl/gl-renderer.h
diff --git a/libweston/renderer-gl/meson.build b/libweston/renderer-gl/meson.build
new file mode 100644
index 00000000..b5d77d58
--- /dev/null
+++ b/libweston/renderer-gl/meson.build
@@ -0,0 +1,37 @@
+if not get_option('renderer-gl')
+	subdir_done()
+endif
+
+config_h.set('ENABLE_EGL', '1')
+
+srcs_renderer_gl = [
+	'gl-renderer.c',
+	linux_dmabuf_unstable_v1_protocol_c,
+	linux_dmabuf_unstable_v1_server_protocol_h,
+]
+
+deps_renderer_gl = [
+	dep_pixman,
+	dep_libweston,
+	dep_libdrm_headers,
+	dep_vertex_clipping
+]
+
+foreach name : [ 'egl', 'glesv2' ]
+	d = dependency(name, required: false)
+	if not d.found()
+		error('gl-renderer requires @0@ which was not found. Or, you can use \'-Drenderer-gl=false\'.'.format(name))
+	endif
+	deps_renderer_gl += d
+endforeach
+
+plugin_gl = shared_library(
+	'gl-renderer',
+	srcs_renderer_gl,
+	include_directories: include_directories('../..', '../../shared'),
+	dependencies: deps_renderer_gl,
+	name_prefix: '',
+	install: true,
+	install_dir: dir_module_libweston
+)
+env_modmap += 'gl-renderer.so=@0@;'.format(plugin_gl.full_path())
-- 
2.21.0

